aspnet/mvc/models/formatting.rst

```diff
@@ -1,8 +1,217 @@
+
+The following action method demonstrates the use of the ``Ok`` and ``NotFound`` helper methods:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 28-38
+  :emphasize-lines: 8,10
+  :dedent: 8
+
+Like the previous examples, this one will return JSON-formatted results in response to requests by default. However, you can use a tool like `Fiddler <http://www.telerik.com/fiddler>`_ to create a request that includes an Accept header, and specify another format. In that case, if the server has a *formatter* that matches the requested format, the results will be returned in that format.
+
+.. image:: formatting/_static/fiddler-composer.png
+
+In the above screenshot, the Fiddler Composer has been used to generate a request, specifying ``Accept: application/xml``. By default, ASP.NET Core MVC only supports JSON, so even when another format is specified, the result returned is still JSON-formatted. You'll see how to add additional formatters in the next section.
+
+Your controller actions can return model objects, in which case ASP.NET MVC will automatically create an ``ObjectResult`` for you that wraps the object. The client will get the appropriately formatted serialized object. If the object being returned is ``null``, then the framework will return a ``204 No Content`` response.
+
+Returning an object type:
+
+.. literalinclude:: formatting/sample/src/ResponseFormattingSample/Controllers/Api/AuthorsController.cs
+  :language: c#
+  :lines: 40-45
+  :emphasize-lines: 3
+  :dedent: 8
+
+In the sample, a request for a valid author alias will receive a 200 OK response with the author's data. A request for an invalid alias will receive a 204 No Content response.
+
+Content Negotiation Process
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Content *negotiation* only takes place if an ``Accept`` header appears in the request. If it does, then the framework will enumerate through the list of formatters until it finds one that can serialize the response object into that format. If no formatter is found that can support the type to be returned, then the framework will send a 406 Not Acceptable response code. However, if the request specifies XML, but the XML formatter has not been configured, then the JSON formatter will be used. Likewise, if no header is given, then the first formatter that can handle the object to be returned will be used to serialize the response (in this case, there isn't any negotiation taking place - the server is determining what format it will use).
```

Comment:
remove Likewise
If no header is specified, the first formatter that can handle the returned object will be used to serialize the response. In this case, there isn’t any negotiation taking place - the server is determining what format it will use.